lib/util.js

```diff
@@ -11,19 +11,20 @@ var Debug;
 
 const formatRegExp = /%[sdj%]/g;
 exports.format = function(f) {
+  const args = arguments;
+  const len  = args.length;
+
   if (typeof f !== 'string') {
-    var objects = [];
-    for (var index = 0; index < arguments.length; index++) {
-      objects.push(inspect(arguments[index]));
+    let objects = new Array(len);
+    for (let i = len; i--;) {
```

Comment:
Conjecture. I've been hearing rumors that let in for loops is not yet fully optimized by v8. Is that rumor incorrect?